sound/soc/sof/intel/hda-ipc.c

```diff
@@ -260,11 +261,20 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
 		snd_sof_dsp_update_bits_unlocked(sdev, HDA_DSP_BAR,
 						 HDA_DSP_REG_ADSPIC,
 						 HDA_DSP_ADSPIC_IPC, 0);
-		ret = IRQ_WAKE_THREAD;
+		spin_unlock(&sdev->hw_lock);
+		sof_ops(sdev)->irq_thread(irq, context);
```

Comment:
I don't get the change here?

Reply:
The sof_ops(sdev)->irq_thread was the irq thread in request_thread_irq(). So we have to call it in the irq thread.